### ISSUE TITLE
Add loading state for discount code retrieval

### ISSUE THREAD

**Issue by jlengstorf** [gatsbyjs/store.gatsbyjs.org#1]
Currently there's a lag when retrieving discount codes. I _think_ this is necessary, because the API has to hit GitHub's API, then Shopify's API, then respond.

However, it currently looks like nothing has happened, which will likely result in duplicate clicks and probably rate limiting issues with GitHub.

We need to:

- [x] disable the button after clicking
- [x] show a loading state
- [x] hide the loading state and re-enable the button (if visible) after the response comes back

**Comment by jlengstorf** [comment]
Moved to public repo: https://github.com/gatsbyjs/store.gatsbyjs.org/issues/8

**Comment by fk** [comment]
Always loved https://lab.hakim.se/ladda/ — maybe https://github.com/jsdir/react-ladda is an option if it's not too heavy?

**Comment by fk** [comment]
> Yeah, I was even thinking of just covering the whole area with a loading state.

Yep that makes sense now that you say it—the user could bail clicking one of the links to GitHub. To prevent the user to go anywhere we'd have to cover the whole screen (also cover the header/navigation) though? (…Smells a little like overthinking (just judging myself here ;-))

> We need to overhaul the form to request it (needs an option to opt in/out of the mailing list, name info, potentially email address), so I need to do that first. 

#49 Lookin' good!

---

Both a "loading screen" component and a "button with loading state indication" are excellent additions to the component library!

**Comment by jlengstorf** [comment]
I didn't add a button with a loading state, but I added form styles to show loading/disable the form during submission.

I tried to make the components smaller and logically connected to it'll be easy to go in and see what's happening. My hope is that you'll only have to play with styles in many cases rather than rewriting components. We'll see how well that goes once you get the prototypes done in #43. 😄
